totalRP3/Modules/NamePlates/NamePlates_Plater.lua

```diff
@@ -0,0 +1,355 @@
+-- Copyright The Total RP 3 Authors
+-- SPDX-License-Identifier: Apache-2.0
+
+local TRP3_API = select(2, ...);
+local TRP3_NamePlates = TRP3_NamePlates;
+local TRP3_NamePlatesUtil = TRP3_NamePlatesUtil;
+local L = TRP3_API.loc;
+
+local Plater = _G["Plater"]
+
+local TRP3_PlaterNamePlates = {};
+
+function TRP3_PlaterNamePlates:OnModuleInitialize()
+	if not IsAddOnLoaded("Plater") then
+		return false, L.NAMEPLATES_MODULE_DISABLED_BY_DEPENDENCY;
+	elseif TRP3_NAMEPLATES_ADDON ~= nil then
+		-- Another nameplate decorator module met its own activation criteria.
+		return false, L.NAMEPLATES_MODULE_DISABLED_BY_EXTERNAL;
+	end
+
+	TRP3_PlaterNamePlates.platerVersion = Plater.versionString
+	TRP3_PlaterNamePlates.platerProfile = Plater.db.profile
+	TRP3_PlaterNamePlates.platerConfig = Plater.db.profile.plate_config
+
+	-- Define TRP3_NAMEPLATES_ADDON now to claim decoration rights. This
+	-- should be sanity checked on OnModuleEnable to make sure someone else
+	-- didn't trample over it.
+
+	TRP3_NAMEPLATES_ADDON = "Plater";
+end
+
+function TRP3_PlaterNamePlates:OnModuleEnable()
+	-- Sanity check TRP3_NAMEPLATES_ADDON to ensure that we remain the chosen
+	-- decorator addon, since it may get trampled by external code.
+
+	if TRP3_NAMEPLATES_ADDON ~= "Plater" then
+		return false, L.NAMEPLATES_MODULE_DISABLED_BY_EXTERNAL;
+	end
+
+	TRP3_NamePlates.RegisterCallback(self, "OnNamePlateDataUpdated");
+
+	TRP3_API.Events.registerCallback("CONFIGURATION_CHANGED", function(...) return self:OnConfigurationChanged(...); end);
+
+	self.unitDisplayInfo = {};
+	self.initialized = {};
+end
+
+function TRP3_PlaterNamePlates:OnConfigurationChanged(_)
+	self:UpdateAllNamePlates()
+end
+
+function TRP3_PlaterNamePlates:OnNamePlateDataUpdated(_, nameplate, unitToken, displayInfo)
+	self:SetUnitDisplayInfo(unitToken, displayInfo);
+	if nameplate.Plater then
+		self:UpdateNamePlate(nameplate);
+	end
+end
+
+function TRP3_PlaterNamePlates:OnNamePlateCreate(nameplate)
+	local nameplateKey = nameplate.unitFramePlater.namePlateUnitToken;
+
+	if self.initialized[nameplateKey] then
+		-- For safety, should be impossible.
+		return;
+	end
+
+	hooksecurefunc(nameplate.unitFramePlater, "UpdateName", function(...) return self:OnNameplateNameTextUpdated(...); end);
+	hooksecurefunc(nameplate.unitFramePlater, "UpdateHealthColor",
+		function(...) return self:UpdateNamePlateHealthBar(...); end);
+
+	hooksecurefunc(Plater, "UpdateUnitName",
+		function(...) return self:UpdateNamePlateNameText(...); end);
+
+	do
+		-- Icon widget.
+		if nameplate.actorType == "friendlyplayer" then
+			local iconWidget = nameplate:CreateTexture("TRP3PlaterIcon", "ARTWORK");
```

Comment:
Drop the "TRP3PlaterIcon" name from this call and replace it with a nil; as-is every nameplate that runs through this function is going to create textures all named the same thing.

Reply:
Fixed in [12b3fa6](https://github.com/Total-RP/Total-RP-3/commit/12b3fa671de0903238b67ebcbc4e764019327305).